vite.config.js

```diff
@@ -27,95 +27,104 @@ const imagesRegex = /images\/.+\.(png|jpe?g|gif|webp|avif|svg|ico)/;
-			}
-			if (imagesRegex.test(filePath)) {
-				return size <= 1; // could be 10000 but we'd need to exclude apple-touch-icons
-			}
-			// vite default
-			return size <= 4096;
-		},
-	},
-	css: {
-		postcss: {
-			plugins: isProd ? [
-				autoprefixer,
-				cssnano,
-				tailwindcss,
-			] : [
-				tailwindcss,
-			],
+export default defineConfig(({ isSsrBuild, mode }) => {
+	return {
+		assetsInclude: [
+			'**/*.bin',
+			'**/*.wasm',
+		],
+		// Force all assets/vite calls through /static in dev mode, compiled mode is covered by the build config below
+		base: isProd ? '/' : '/static/',
+		// Use /static for all assets in prod mode
+		build: {
+			assetsDir: 'static',
+			assetsInlineLimit: (filePath, content) => {
+				const size = content.length;
+				if (fontsRegex.test(filePath) || mediaRegex.test(filePath)) {
+					return size <= 10000;
+				}
+				if (imagesRegex.test(filePath)) {
+					return size <= 1; // could be 10000 but we'd need to exclude apple-touch-icons
+				}
+				// vite default
+				return size <= 4096;
+			},
 		},
-		preprocessorOptions: {
-			scss: {
+		css: {
+			postcss: {
+				plugins: isProd ? [
+					autoprefixer,
+					cssnano,
+					tailwindcss,
+				] : [
+					tailwindcss,
+				],
+			},
+			preprocessorOptions: {
+				scss: {
 				// Suppress deprecation warnings from node modules
-				quietDeps: true,
+					quietDeps: true,
+				},
 			},
 		},
-	},
-	define: {
-		UI_COMMIT: JSON.stringify(gitRevisionPlugin.commithash()),
-		UI_BRANCH: JSON.stringify(gitRevisionPlugin.branch()),
-		UI_TAG: JSON.stringify(gitRevisionPlugin.version()),
-	},
-	plugins: [
-		vue({
-			template: {
-				compilerOptions: {
-					compatConfig: {
-						MODE: 3,
+		define: {
+			UI_COMMIT: JSON.stringify(gitRevisionPlugin.commithash()),
+			UI_BRANCH: JSON.stringify(gitRevisionPlugin.branch()),
+			UI_TAG: JSON.stringify(gitRevisionPlugin.version()),
+		},
+		plugins: [
+			vue({
+				template: {
+					compilerOptions: {
+						compatConfig: {
+							MODE: 3,
+						},
 					},
 				},
+			}),
+			// load .graphql and .gql files
+			graphQLLoader(),
+			// load svg files as vue components
+			svgLoader({
+				svgoConfig: {
+					plugins: [
+						{ name: 'removeTitle', active: false },
+					],
+				},
+			}),
+			// svg icon sprite sheet
+			svgStore({
+				dirs: ['src/assets/icons/sprite'],
+				symbolId: 'icon-[name]',
+				optimizeOptions: {
+					floatPrecision: 3,
+				},
+			}),
+		],
+		resolve: {
+			alias: {
+				// alias src directory
+				'#src': resolve('src'),
+				// alias promise module to handle timesync calling require('promise')
+				promise: resolve('build/promise.js'),
+				// this alias is required for the rendering of src/components/Contentful/DynamicRichText.vue
+				// it should only be used on the client build in production
+				// eslint-disable-next-line max-len
+				...(isSsrBuild === false && mode === 'production' && { vue: path.resolve(__dirname, 'node_modules', 'vue', 'dist', 'vue.esm-bundler.js') })
```

Comment:
This is the main change in this PR. 
We have to use 'vue' as the alias target, and the 'vue.esm-bundler.js' build. And we can only have this alias for the client side production build.